src/phoenix/session/session.py

```diff
@@ -620,9 +621,15 @@ def _is_colab() -> bool:
 def _is_sagemaker() -> bool:
     """Determines whether this is in a SageMaker notebook"""
     try:
-        import sagemaker  # type: ignore # noqa: F401
+        with open(os.devnull, "w") as devnull:
+            with redirect_stdout(devnull):
+                import sagemaker  # type: ignore # noqa: F401
```

Comment:
nit. It's probably simpler to just dump it to a string and garbage collect it. Using devnull involves an unnecessary system call and OS context switching.
```suggestion
        with redirect_stdout(io.StringIO()):
            import sagemaker  # type: ignore # noqa: F401
```

Reply:
or maybe find_spec is better
```suggestion
       if find_spec("sagemaker") is None:
```

Reply:
addressed in [127f6dd](https://github.com/Arize-ai/phoenix/pull/2527/commits/127f6ddac331ac013638021def7199297b56e15f)